socorro/processor/cache_manager.py

```
@@ -293,15 +307,45 @@ def _event_generator(self, nonblocking=False):
                             # Handle directory events which update our watch lists
                             if flags.CREATE & event_mask:
                                 try:
-                                    self.add_watch(path)
+                                    created_wd = self.add_watch(path)
                                 except OSError:
+                                    logger.exception("add watch error for %s", path)
```

Adding a watch should never silently fail--that was a mistake. :(